### PR TITLE
Sp/session mixin typehint support

### PR DESCRIPTION
**Pull request by technicalgrp89**
type annotations support
mypy compatible code
utils.groovy - mypy cmd added

### CODE REVIEW

File: tuna/session_mixin.py [hunk 1]
```diff
@@ -82,7 +88,7 @@ def insert_session(self):
         LOGGER.warning("Err occurred trying to add new session: %s \n %s", err,
                        self)
         session.rollback()
-        entry = self.get_query(session, type(self), self).one()
+        entry = self.get_query(session, type(self), self).one()  #type: ignore
```
Comment (alexandraBara) [inline tuna/session_mixin.py]: why is this ignored? this should be of type SessionExample or Session.

File: tuna/session_mixin.py [hunk 2]
```diff
@@ -25,29 +25,35 @@
...
 from tuna.utils.logger import setup_logger
 
-LOGGER = setup_logger('session')
+LOGGER: logging.Logger = setup_logger('session')
 
 
 class SessionMixin():
   """Session Mixin to provide interface for the session table"""
   #pylint: disable=too-few-public-methods
+  #pylint: disable=too-many-instance-attributes
```
Comment (alexandraBara) [inline tuna/session_mixin.py]: is this needed? Have you tried removing it? doesnt seem like that many instance attributes

**Review by alexandraBara — CHANGES_REQUESTED** [review]

File: tuna/session_mixin.py (hunk 2 above)
Comment (technicalgrp89) [reply tuna/session_mixin.py]: removed

File: tuna/session_mixin.py (hunk 1 above)
Comment (technicalgrp89) [reply tuna/session_mixin.py]: overload function defiition added

**Review by alexandraBara — APPROVED** [review]

**Review by iq136boy — APPROVED** [review]

**Review by reidkwja — APPROVED** [review]

File: requirements.txt [hunk 3]
```diff
@@ -38,7 +38,7 @@ py==1.10.0
 pyasn1==0.4.4
 pycparser==2.19
 Pygments==2.14.0
-pylint>=2.14.5
+pylint<=2.17.0-dev0,>=2.15.4
```
Comment (cderb) [inline requirements.txt:41]: I'm guessing this is the range returned by the error log. Should we be using the range or just >=2.15.4?

**Review by cderb — APPROVED** [review]

**Pull request merged**